app/uk/gov/hmrc/cataloguefrontend/connector/ConfigConnector.scala

```diff
@@ -46,4 +48,15 @@ class ConfigConnector @Inject() (
     implicit val brsr = BobbyRuleSet.reads
     http.GET[BobbyRuleSet](s"$serviceConfigsBaseUrl/bobby/rules")
   }
+
+  def deploymentConfig(
+    service: String,
+    environment: Environment
+  )(implicit hc: HeaderCarrier): Future[Option[DeploymentConfig]] = {
+    implicit val dcr = DeploymentConfig.reads
+    http
+      .GET[DeploymentConfig](s"$serviceConfigsBaseUrl/deployment-config/${environment.asString}/$service")
+      .map(Some(_))
+      .recover { case _ => None }
```

Comment:
This is quite blunt. Would we prefer to log the error here before returning `None`?

Reply:
I think it should at least log the error.
This function could be used in other contexts, so I think propagating the error is best - the api is internal, and we don't expect it to be down. `None` usually means there is no config, so it could be confusing. - note, if we're expecting to handle a 404 as `None` it should be `.GET[Option[DeploymentConfig]]` - the implicit reads will handle the 404.

Reply:
Ah, yes, `.GET[Option[DeploymentConfig]]` is exactly what I want here, thanks